src/integrations/kafka/consumer.hpp

```diff
@@ -137,6 +140,13 @@ class Consumer final : public RdKafka::EventCb {
   /// Returns true if the consumer is actively consuming messages.
   bool IsRunning() const;
 
+  /// Sets the consumer's offset.
+  ///
+  /// This function returns the empty string on success or an error message otherwise.
+  ///
+  /// @param offset: the offset to set.
+  std::string SetConsumerOffsets(int64_t offset);
```

Comment:
I would change this to `[[nodiscard]] utils::BasicResult<std::string>`. It is more clear even without documentation and enforces the error check.